### DISCUSSION TITLE
Able to connect to JS socket server v4.2.1?

### DISCUSSION THREAD

**Discussion by sitihawasabri** [doghappy/socket.io-client-csharp#240] [Q&A]
Hi, I tried to connect to JS socket server, it's version 4.2.1 but failed. This is the error which it returns me;

_**An error has occurred during a TLS handshake. ---> System.Security.Authentication.AuthenticationException: A call to SSPI failed, see inner exception. ---> System.ComponentModel.Win32Exception: The function requested is not supported.**_

I think it may be version issue, because last time I was able to connect to JS socket server with no issue.

**Comment by doghappy** [comment]
Please check this issue: https://github.com/doghappy/socket.io-client-csharp/issues/236

**Reply by sitihawasabri** [reply]
I have commented on that issue and I faced this issue; 

![image](https://user-images.githubusercontent.com/53597690/144558813-b774d97d-3bd2-4511-ae30-669dc3ee7227.png)

'SocketIO' does not contain a definition for 'ClientWebSocketProvider' and no accessible extension method
'ClientWebSocketProvide' accepting a first argument of type 'SocketIO' could be found (are you missing a using directive or an assembly reference?)

May I know which SocketIOClient.NetFx version should I use? I currently use the latest version: 2.0.2.10. Or I did miss some parts?

**Reply by doghappy** [reply]
Are you based on `.NET Core` or `.NET Framework`?

**Reply by sitihawasabri** [reply]
.NET Framework

**Reply by doghappy** [reply]
Have you tried the following code?

```cs
    System.Net.ServicePointManager.ServerCertificateValidationCallback = (sender, certificate, chain, sslPolicyErrors) =>
    {
        Console.WriteLine(sslPolicyErrors);
        // Note it will ignore all SSL errors
        return true;
    };
```

**Reply by sitihawasabri** [reply]
Yes, but the issue is at L51; **client._ClientWebSocketProvider_** 

![image](https://user-images.githubusercontent.com/53597690/144560491-c1d2eeab-4f7b-4765-8c17-324572da5829.png)

Do I need to include anything else before L51?

**Reply by doghappy** [reply]
The code you provided is wrong 😂

~client.ClientWebSocketProvider~

please use the following code instead

```cs
    System.Net.ServicePointManager.ServerCertificateValidationCallback = (sender, certificate, chain, sslPolicyErrors) =>
    {
        Console.WriteLine(sslPolicyErrors);
        // Note it will ignore all SSL errors
        return true;
    };
```

**ClientWebSocketProvider is not included in the code**

---

ClientWebSocketProvider is provided by the new version, the version you are using does not have this api

**Reply by sitihawasabri** [reply]
Really? Wait I don't quite get it yet. Can you send me the link for this one, full code? Where do I put the socket server URL? 

_ClientWebSocketProvider is provided by the new version, the version you are using does not have this API_ -- Meaning if I want to use this ClientWebSocketProvider, I need to update to which version?

**Reply by doghappy** [reply]
Just set it up before connecting

full code:

```cs
    class Program
    {
        static async Task Main(string[] args)
        {
            System.Net.ServicePointManager.ServerCertificateValidationCallback = (sender, certificate, chain, sslPolicyErrors) =>
            {
                Console.WriteLine(sslPolicyErrors);
                // Note it will ignore all SSL errors
                return true;
            };

            var uri = new Uri("http://localhost:11003/");

            var socket = new SocketIO(uri, new SocketIOOptions
            {
                    ....
            });

           await socket.ConnectAsync();
    }
```

**Reply by sitihawasabri** [reply]
Okay noted, let me try first. Thank you so much for your prompt reply! ☺️

**Reply by sitihawasabri** [reply]
Hi @doghappy , after I tried using this  

System.Net.ServicePointManager.ServerCertificateValidationCallback = (sender, certificate, chain, sslPolicyErrors) =>
    {
        Console.WriteLine(sslPolicyErrors);
        // Note it will ignore all SSL errors
        return true;
    };

I still got returned SSL errors; 

**WebSocket.connect|WebSocketSharp.WebSocketException: An error has occurred during a TLS handshake. ---> System.ArgumentException: The specified value is not valid in the 'SslProtocolType' enumeration.
                             Parameter name: sslProtocolType
                                at System.Net.Security.SslState.ValidateCreateContext(Boolean isServer, String targetHost, SslProtocols enabledSslProtocols, X509Certificate serverCertificate, X509CertificateCollection clientCertificates, Boolean remoteCertRequired, Boolean checkCertRevocationStatus, Boolean checkCertName)
                                at System.Net.Security.SslStream.AuthenticateAsClient(String targetHost, X509CertificateCollection clientCertificates, SslProtocols enabledSslProtocols, Boolean checkCertificateRevocation)
                                at WebSocketSharp.WebSocket.setClientStream()**
                                
 I have tried with Tls, Tls11, Tls12 but still getting the same error. Is there any way I can solve this issue?
